src/pages/Detail.tsx

```diff
@@ -0,0 +1,30 @@
+import { ProductService } from '../service';
+import { useEffect, useState } from 'react';
+import { IProduct } from '../types/shoppingCart';
+import { useParams } from 'react-router-dom';
+import ProductDetail from '../components/organisms/product/ProductDetail';
+import { CardDetail } from '../components/template';
+
+export default function Detail() {
+  const { findProductOne } = ProductService();
+  const { id } = useParams();
+  const [product, setProduct] = useState<IProduct>();
+
+  const fetchData = async () => {
+    const item = await findProductOne(Number(id));
+
+    setProduct(item);
+  };
+
+  useEffect(() => {
+    fetchData();
+  }, []);
+
+  if (!product) return;
```

Comment:
```suggestion
  if (!product) return null;
```

undefined 는 유효한 리액트 컴포넌트의 반환값이 아닙니다~
아무것도 렌더링하지 않는걸 나타내려면 `null`을 반환해주세요!